library/Freckle/App/Stats.hs

```diff
@@ -108,17 +126,26 @@ withStatsClient
   -> (StatsClient -> m a)
   -> m a
 withStatsClient StatsSettings {..} f = do
+  gauges <- liftIO $ fmap Gauges EKG.new
   if amsEnabled
     then do
       tags <- (amsTags <>) <$> getEcsMetadataTags
-
       Datadog.withDogStatsD amsSettings $ \client ->
         -- Add the tags to the thread context so they're present in all logs
         withThreadContext (map toPair tags)
-          $ f StatsClient { scClient = client, scTags = tags }
-    else f $ StatsClient { scClient = Datadog.Dummy, scTags = amsTags }
+          $ f StatsClient { scClient = client, scTags = tags, scGauges = gauges }
+    else do
+      f $ StatsClient { scClient = Datadog.Dummy, scTags = amsTags, scGauges = gauges }
   where toPair = bimap (fromString . unpack) String
 
+withGauge :: (MonadReader app m, HasStatsClient app, MonadUnliftIO m) => Lens' Gauges EKG.Gauge -> m a -> m a
```

Comment:
This still leaks `EKG.Gauge` in a user-facing type signature. We're so close to _not_ doing that, I think it's worth one last push.

What do you think of:

```hs
module Freckle.App.Stats
  ( ...

  -- * Stateful Gauges
  , GaugeName
  , dbConnections
  , withGauge

  -- * Reporting
  , ...
  ) where

newtype GaugeName = GaugeName
  { unGaugeName :: Gauges -> EKG.Gauge
  }

dbConnections :: GuageName
dbConnections = GaugeName gDBConnections -- rename the actual record accessor

withGauge :: ... => GaugeName -> m a -> m a
withGauge guageName = do
  gauge' <- view $ statsClientL . guagesL . to (unGaugeName guageName)
```

Usage is identical, internals aren't any more complex, and the interface is better (IMO) because we don't leak `EKG.` or inflict our users with `Lens`.

Thoughts?